cmd/root.go

```diff
@@ -54,17 +52,6 @@ var libbeatConfigOverrides = func() []cfgfile.ConditionalOverride {
 				},
 			}),
 		},
-		{
-			Check: func(_ *agentconfig.C) bool {
-				return os.Getenv(cloudEnv) != ""
```

Comment:
Is this flag used to check if APM Server is running on cloud?

Reply:
Yes, it is set only on Cloud. Its value is the memory size of the APM instance. We used to use it for configuring queue sizes, and number of goroutines for the libbeat Elasticsearch output. When we switched to our new modelindexer implementation, that became unnecessary: https://github.com/elastic/apm-server/pull/6656/files#diff-ab967ab1a2f3a1b769106eeb7bfe892ef0e81d1d27811fa15be08e6749feee1fL123